core/trino-main/src/test/java/io/trino/connector/TestingTableFunctions.java

```diff
@@ -1235,6 +1237,7 @@ public static ConnectorSplitSource getConstantFunctionSplitSource(ConstantFuncti
             if (remainingSize > 0) {
                 splits.add(new ConstantFunctionSplit(remainingSize));
             }
+            splits.add(new EmptySplit(TEST_CATALOG_HANDLE));
```

Comment:
Why don't we test with a function which has an empty split source? An `EmptySplit` should be produced by `SourcePartitionedScheduler`, and captured by the Operator. 
To achieve it, you could slightly modify the testing function `ConstantFunction`: accept `0` as the argument `count`, and produce zero splits in that case.